database/src/lib.rs

```diff
@@ -434,11 +434,11 @@ pub struct LabelId(pub u8, pub u32);
 #[derive(Serialize, Deserialize, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
 pub struct ArtifactIdNumber(pub u32);
 
+/// Cached results of various queries.
 #[derive(Debug, Clone, PartialEq, Eq, Default)]
 pub struct Index {
     commits: Indexed<Commit>,
```

Comment:
I'd like to add comments to these fields, but I'm still not exactly sure what they are. For instance, commits seems to be from the `artifact` table with `type` master or try. I'm not exactly sure what that should be.

Reply:
Can you say more about your confusion? These fields are *intended* to be essentially duplicates of the various tables (artifacts, error_series, pstat_series, etc.) so that we can avoid a network round-trip in the case of Postgres when fetching the ID used in the "main" table.

To some extent I think the presence of these is mostly an artifact of not wanting to write too much SQL -- it's much easier to have ad-hoc searching in Rust code than threading through SQL queries. In practice IIRC it was also a little faster, but whether that's still true or if I'm remembering right I'm less sure.

Reply:
My comment is a bit old as I know understand more about the structure. Perhaps we should do a short write up on the database structure. We could then refer to that here.